packages/earn-controller/src/EarnController.ts

```diff
@@ -303,7 +303,7 @@ export class EarnController extends BaseController<
     messenger: EarnControllerMessenger;
     state?: Partial<EarnControllerState>;
     addTransactionFn: typeof TransactionController.prototype.addTransaction;
-    selectedNetworkClientId: string;
+    selectedNetworkClientId: () => string;
```

Comment:
Hmm, this is not something we typically have to do in any other controller, so this is surprising to me that we have to do this. In your PR description you mention that it is necessary after the controller is reinitialized. When would this occur? I know that the Engine gets reinitialized in mobile in certain cases but usually _all_ controllers get reinitialized. 

If we still think this is necessary, have you considered moving the asynchronous operations in this controller to an `init` function? You should then be able to get the `selectedNetworkClientId` from NetworkController by going through its state.

Reply:
Either an `init` method in EarnController or using the init pattern on the client side should work, whichever makes sense.